library/foreman_provisioner.py

```diff
@@ -0,0 +1,292 @@
+
+        headers = {'Accept': 'application/json',
+                   'Content-type': 'application/json'}
+
+        if extra_headers:
+            headers.update(extra_headers)
+
+        self.session.headers.update(headers)
+
+        self.url = url.rstrip('/')
+        self.default_uri = '/api/v2/hosts/'
+
+    def reserve_host(self, host_id):
+        """
+        This method 'tags' a host as reserved in foreman
+        :param host_id: the name of ID of the host we wish to reserve
+        :returns: the host information on success, else empty body
+        :rtype: list of dictionaries -- [{"host": {}}]
+        """
+        #TODO(tkammer): add the option to provide the query itself after "?"
+        request_url = '{0}/api/hosts_reserve' \
+                      '?query=name ~ {1}'.format(self.url, host_id)
+        response = self.session.get(request_url, verify=False)
+        body = response.json()
+        return body
+
+    def release_host(self, host_id):
+        """
+        This method removed the 'tag' made by 'reserve_host" in foreman
+        :param host_id: the name or ID of the host we wish to release
+        :returns: the host name
+        :rtype: list of strings
+        """
+        request_url = '{0}/api/hosts_release' \
+                      '?query=name ~ {1}'.format(self.url, host_id)
+        response = self.session.get(request_url, verify=False)
+        body = response.json()
+        Exception(body)
+        return body
+
+    def get_host(self, host_id):
+        """
+        This method returns the host details as listed in the foreman
+        :param host_id: the name or ID of the host we wish to get
+        :returns: host information
+        :rtype: dict
+        """
+        request_url = '{0}/{1}/{2}'.format(self.url, self.default_uri, host_id)
+        response = self.session.get(request_url, verify=False)
+        body = response.json()
+        return body
+
+    def update_host(self, host_id, update_info):
+        """
+        This method updates a host details in foreman
+        :param host_id: the name or ID of the host we wish to update
+        :param update_info: params we wish to update on foreman
+        :type update_info: dict
+        :returns: host information
+        :rtype: dict
+        """
+        request_url = '{0}/{1}/{2}'.format(self.url, self.default_uri, host_id)
+        response = self.session.put(request_url, data=update_info, verify=False)
+        body = response.json()
+        return body.get('host')
+
+    def set_build_on_host(self, host_id, flag):
+        """
+        sets the 'build' flag of a host to a given :param flag:
+        :param host_id: the id or name of the host as it is listed in foreman
+        :param flag: a boolean value (true/false) to set the build flag with
+        """
+        host = self.update_host(host_id, json.dumps({'build': flag}))
+        #TODO(tkammer): add verification that the host was updated
```

Comment:
Please handle this comment and remove it.
The handle is probably should be in the 'update_host' method - just compare the details before and after the update